content/shared/components/solid.ts

```diff
@@ -154,23 +106,74 @@ export class Solid extends CustomElementView {
     this.scene.draw();
   }
 
+  private handleRotation(animate = true, speed = 1) {
+    // TODO Damping after mouse movement
+    // TODO Better mouse-to-point mapping
+
+    // Only Chrome is fast enough to support auto-rotation.
+    const autoRotate = animate && Browser.isChrome && !Browser.isMobile;
+    this.css('cursor', 'grab');
+
+    let dragging = false;
+    let visible = false;
+
+    const frame = () => {
+      if (visible && autoRotate) requestAnimationFrame(frame);
+      this.scene.draw();
+      if (!dragging) this.object.rotation.y += speed * 0.012;
+    };
+
+    if (autoRotate) {
+      this.scene.$canvas.on('enterViewport', () => {
+        visible = true;
+        frame();
+      });
+      this.scene.$canvas.on('exitViewport', () => visible = false);
+    } else {
+      setTimeout(frame);
+    }
+
+    // The 1.1 creates rotations that are slightly faster than the mouse/finger.
+    const s = Math.PI / 2 / this.scene.$canvas.width * 1.1;
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    slide(this.scene.$canvas, {
+      start() {
+        dragging = true;
+        $html.addClass('grabbing');
+      },
+      move(posn, start, last) {
+        const d = posn.subtract(last).scale(s);
+        const q = new THREE.Quaternion().setFromEuler(new THREE.Euler(d.y, d.x));
+        self.object.quaternion.multiplyQuaternions(q, self.object.quaternion);
+        self.trigger('rotate', {quaternion: self.object.quaternion});
+        if (!autoRotate) frame();
+      },
+      end() {
+        dragging = false;
+        $html.removeClass('grabbing');
+      }
+    });
+  }
+
 
   // ---------------------------------------------------------------------------
   // Element Creation Utilities
 
-  addLabel(text: string, posn: Vector, color = STROKE_COLOR, margin = '') {
+  addLabel(text: string, posn: Vector, color = STROKE_COLOR, margin = [0, 0]) {
```

Comment:
This will break some of the existing code. `margin` is a CSS property, so you can already specify different x/y coordinates, e.g. as `3px 5px`